debian/control

```diff
@@ -20,7 +20,7 @@ Depends: open62541 (= ${binary:Version}), ${misc:Depends}
 Description: Development header files for open62541
  open62541 is an open source C (C99) implementation of the OPC UA standard
 
-Package: open62541
+Package: libopen62541-1
```

Comment:
This should be automated. If we change the major version, that part should be updated correspondingly.
We should try to integrate this into the python script from PR #2980
https://github.com/open62541/open62541/pull/2980/files#diff-c83ebe4f93ca1086557dbd0e52e2ae3c

Otherwise creating new releases will be error prone and one might forget to update the version here.

Reply:
> This should be automated. If we change the major version, that part should be updated correspondingly.
> We should try to integrate this into the python script from PR #2980
> https://github.com/open62541/open62541/pull/2980/files#diff-c83ebe4f93ca1086557dbd0e52e2ae3c
> 
> Otherwise creating new releases will be error prone and one might forget to update the version here.
> 
> Maybe we can have a `debian/control.template` which is then read by the python script and updated correspondingly. That script then also renames the file `debian/libopen62541.template` correspondingly.

Yes, I agree, would be nice to have!

AFAIK this unfortunately doesn't work building the package the debian way (i.e. with pbuilder etc.) because  `debian/control` and `debian/changelog` will be referenced before `cmake` will be called!?
But for sure we can use a template for the lib-install-file  `debian/lib62541-<soname>.install` !

Reply:
Hmm, yes, right... For the launchpad builds we may need to update the recipe to execute the python script, if that's even possible. Let's merge #2980 first, then we can have a follow-up fix which includes this PR with the comments

Reply:
> > This should be automated. If we change the major version, that part should be updated correspondingly.
> > We should try to integrate this into the python script from PR #2980
> > https://github.com/open62541/open62541/pull/2980/files#diff-c83ebe4f93ca1086557dbd0e52e2ae3c
> > Otherwise creating new releases will be error prone and one might forget to update the version here.
> > Maybe we can have a `debian/control.template` which is then read by the python script and updated correspondingly. That script then also renames the file `debian/libopen62541.template` correspondingly.
> 
> Yes, I agree, would be nice to have!
> 
> AFAIK this unfortunately doesn't work building the package the debian way (i.e. with pbuilder etc.) because `debian/control` and `debian/changelog` will be referenced before `cmake` will be called!?
> But for sure we can use a template for the lib-install-file `debian/lib62541-<soname>.install` !

BTW: the nice side effect using a template for `debian/lib62541-<soname>.install` is that an error occurs during debian build if the `debian/control` doesn't correspond to the generated `debian/lib62541-<soname>.install`. This solves the problem if one forget  to update `debian/changelog` and `debian/control` in some way when the major version changes. Unfortunately AFAIK one has to correct `debian/changelog` and `debian/control` manually.